src/dioptra/restapi/v1/plugin_parameter_types/service.py

```diff
@@ -533,3 +535,59 @@ def create_all(
             )
             for new_builtin_parameter_type in new_builtin_parameter_types
         ]
+
+
+def get_plugin_task_parameter_types_by_id(
+    ids: Iterable[int], log: BoundLogger
+) -> dict[int, models.PluginTaskParameterType]:
+    """Gets all of the PluginTaskParameterType instances based on the given ids
+    iterable
+
+    Args:
+        ids: an iterable containing all of the ids
+        log: where log messages should go
+
+    Returns:
+        The a dictionary of ids to PluginTaskParameterType instances
+
+    Raises:
+        ValueError: if ids does not contain at least one value
+        EntityDoesNotExistError: if one or more of the ids does not exist in the
+            PluginTaskParameterType table
+    """
+    id_list = set(ids)
+    num_ids = len(id_list)
+    if num_ids < 1:
+        return {}
+    parameter_types_stmt = (
+        select(models.PluginTaskParameterType)
+        .join(models.Resource)
+        .where(
+            models.PluginTaskParameterType.resource_id.in_(tuple(id_list)),
+            models.Resource.is_deleted == False,  # noqa: E712
+            models.Resource.latest_snapshot_id
+            == models.PluginTaskParameterType.resource_snapshot_id,
+        )
+    )
+    parameter_types = db.session.scalars(parameter_types_stmt).all()
+
+    if parameter_types is None:
+        log.error(
+            "The database query returned a None when it should return plugin "
+            "parameter types.",
+            sql=str(parameter_types_stmt),
+            num_expected=num_ids,
+        )
+        raise BackendDatabaseError
+
+    if not len(parameter_types) == num_ids:
+        returned_parameter_type_ids = set([x.resource_id for x in parameter_types])
+        ids_not_found = id_list - returned_parameter_type_ids
+        raise EntityDoesNotExistError(
+            "plugin task parameter types",
+            num_expected=num_ids,
+            num_found=len(parameter_types),
+            ids_not_found=sorted(list(ids_not_found)),
+        )
+
+    return {x.resource_id: x for x in parameter_types}
```

Comment:
We should try and integrate this with the repository class that's been created for plugin parameter types. I'll need to check a little more on how to do it, so leaving this as a note for now.

I'm noticing this isn't used in the service class itself, so it's a utility being called somewhere else. It's basically a read-only view, but I see that it's used in a couple different places where the `uow` class needed to access the repositories won't be available yet. So we might not be able to fully get rid of this, but I don't think this file is the right spot for this as a temporary location.

Reply:
We can punt on where this function should live for now and leave it where it is. We can re-evaluate after the repo pattern is done.